### PR TITLE
Reland: `mul`: convert inputs to result type.

### PR DESCRIPTION
**Pull request by ysiraichi**
Fix: #7266 

This PR tries to re-land #7130, which was reverted due to regressions on XLA:TPU. On top of that, it also fixes a subtle bug that went under the radar. Although we couldn't find a minimal reproducer, internal CI for TPU was reportedly broken due to this PR.

 Therefore, the idea is to have this PR merged, and check the internal CI for TPU results.

cc @miladm @vanbasten23 @JackCaoG

### CODE REVIEW

File: torch_xla/csrc/aten_xla_type.cpp [hunk 1]
```diff
@@ -158,7 +158,7 @@ class OpConfig {
         inputs.begin(), inputs.end(), [](const at::Tensor& tensor) {
           return bridge::TryGetXlaTensor(tensor);
         });
-    XLA_CHECK(it != inputs_.end());
+    XLA_CHECK(it != inputs.end());
```
Comment (ysiraichi) [inline torch_xla/csrc/aten_xla_type.cpp:161]: Previously, incorrectly checking against `inputs_` instead of `inputs`.

Comment (JackCaoG) [reply torch_xla/csrc/aten_xla_type.cpp:161]: man! I should catch this during reviewing haha

**Review by JackCaoG — APPROVED** [review]

File: torch_xla/csrc/aten_xla_type.cpp (hunk 1 above)
Comment (ysiraichi) [reply torch_xla/csrc/aten_xla_type.cpp:161]: Must admit that it was the best naming choice, though.

Comment (JackCaoG) [reply torch_xla/csrc/aten_xla_type.cpp:161]: nice catch 🤣 , I will check the TPU CI after this pr is merged.

Comment (ysiraichi) [reply torch_xla/csrc/aten_xla_type.cpp:161]: Again, sorry for the typo there. That was really bad.

**Pull request merged**